core/src/main/java/org/jdbi/v3/core/Handle.java

```diff
@@ -651,8 +651,14 @@ public TransactionIsolationLevel getTransactionIsolationLevel() {
                 .orElseThrow(() -> new NoSuchExtensionException("Extension not found: " + extensionType));
     }
 
-    public <C extends ExtensionConfig<C>> void configureExtension(Class<C> configClass, Consumer<C> consumer) {
+    public <C extends ExtensionConfig<C>> Handle configureExtension(Class<C> configClass, Consumer<C> consumer) {
         config.extensionRegistry.configure(configClass, consumer);
+        return this;
+    }
+
+    public <C extends ExtensionConfig<C>> Handle configure(Class<C> configClass, Consumer<C> consumer) {
+        registerExtension(new ConfigOnlyExtension<C>(configClass));
+        return configureExtension(configClass, consumer);
```

Comment:
If we're looking for arbitrary configuration storage, then it probably makes sense to separate that out from extensions. That is, extensions can have configuration, but they can fetch whatever configuration they're interested in, rather than expecting core to understand what each extension needs.